### PR TITLE
Fix primitive class handling in Serialisers

### PR DESCRIPTION
**Pull request by geoand**
- Fixes: ##37776

### CODE REVIEW

**Review by franz1981 — APPROVED** [review]
I prefer this one as well!!! Thanks @geoand

**Comment by geoand** [comment]
🙏

File: independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/resource/basic/resource/ResponseInfoResource.java
Comment (gsmet) [inline independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/resource/basic/resource/ResponseInfoResource.java:21]: Should we add a test instead of changing an existing one?

Or this is safe?

Comment (geoand) [reply independent-projects/resteasy-reactive/server/vertx/src/test/java/org/jboss/resteasy/reactive/server/vertx/test/resource/basic/resource/ResponseInfoResource.java:21]: It's totally safe because the response itself was not being used anywhere and now it is.

I also checked that with this test the previous version of the code was failing

**Comment by franz1981** [comment]
@geoand please check if there was some useless private empty method which was used in the Serialisers https://github.com/quarkusio/quarkus/pull/37778/files#diff-4331c4565452796771e5f021cc452593f1cbb4ee69557f63868aae37641769c6L263

**Comment by geoand** [comment]
There don't seem to be any

**Comment by franz1981** [comment]
I am surprised...the diff I have posted was with my change that was removing that dangerous used empty method...what it has gone?

**Review by michalvavrik — APPROVED** [review]
I've checked and this fixes my issue.

**Comment by geoand** [comment]
@franz1981 thanks! Fixed

**Review by franz1981 — APPROVED** [review]
LGTM!

**Comment by michalvavrik** [comment]
Would it be possible to get it in, please?

**Comment by michalvavrik** [comment]
> geoand requested a review from gsmet 10 hours ago

ah, I didn't mention of this, okay let's wait, just annoying to have CI failures

**Comment by geoand** [comment]
Yeah, let's wait for a review from @gsmet

**Comment by michalvavrik** [comment]
> Yeah, let's wait for a review from @gsmet

Sure, np, I'll disable tests in out TS and we can enable them when @gsmet find a time to review. Thanks for the fix.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Sorry for the delay!

**Comment by geoand** [comment]
🙏

**Pull request merged**
